tests/e2e/multi_attach.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	awscloud "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/cloud"
+	ebscsidriver "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/driver"
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/tests/e2e/testsuites"
+	. "github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = Describe("[ebs-csi-e2e] [single-az] [multi-attach] Multi-Attach", func() {
```

Comment:
Can you explain what's going on with the VolumeBindingMode changes? For the failure cases, why do we care whether it's `WaitForFirstConsumer` or not? More importantly, why aren't you testing `WaitForFirstConsumer` in the success case?

Reply:
If no `VolumeBindingMode` mode is defined then we fall back to immediate binding mode but we want to intentionally test dynamic provisioning here. These `VolumeBindingMode` changes /  logic has been revised in latest revision.

Reply:
To provide more context, `SetupDynamicPersistentVolumeClaim` waits for the PVC to be bound if the binding mode is not specified or is set to Immediate. The resulting failures modes here for MA are unbound PVCs and thus the volume is never provisioned, ex: 

```
Events:
  Type     Reason                Age                From                                                                                      Message
  ----     ------                ----               ----                                                                                      -------
  Normal   WaitForFirstConsumer  52s                persistentvolume-controller                                                               waiting for first consumer to be created before binding
  Normal   Provisioning          21s (x6 over 52s)  ebs.csi.aws.com_ebs-csi-controller-7546559f47-wbzxj_7b2c2cef-1129-4330-b8c1-fd9a1de54739  External provisioner is provisioning volume for claim "ebs-4152/pvc-frsm5"
  Warning  ProvisioningFailed    21s (x6 over 52s)  ebs.csi.aws.com_ebs-csi-controller-7546559f47-wbzxj_7b2c2cef-1129-4330-b8c1-fd9a1de54739  failed to provision volume with StorageClass "ebs-4152-ebs.csi.aws.com-dynamic-sc-ns8v8": rpc error: code = Internal desc = Could not create volume "pvc-e7ddd0ea-f26a-4851-a297-fab9761de805": CreateDisk: multi-attach is only supported for io2 volumes
```

> why do we care whether it's WaitForFirstConsumer or not? More importantly, why aren't you testing WaitForFirstConsumer in the success case?

It doesn't fundamentally alter the test's validity, we specify the binding mode in the failure cases due to the quirk mentioned above with `SetupDynamicPersistentVolumeClaim` also validating the state of the PVC.

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/02f1d73eda27571e4e8155d451308a9916de7f1a/tests/e2e/testsuites/specs.go#L140